test/process/extract/ExtractorTest.cpp

```diff
@@ -0,0 +1,350 @@
+
+        bool lastBlockEndedWithNewline = true;
+        Indexer indexer(make_shared<PathSource>(fastq), make_shared<PathSink>(index),
+                        make_shared<BlockDistanceStorageStrategy>(1, true), true);
+        for (auto bd : _blockData) {
+            auto split = StringHelper::splitStr(bd);
+            bool currentBlockEndedWithNewline;
+            u_int32_t numberOfLinesInBlock;
+            int64_t offset = 0;
+            auto entry = indexer.createIndexEntryFromBlockData(bd, split, offset, lastBlockEndedWithNewline,
+                                                               &currentBlockEndedWithNewline,
+                                                               &numberOfLinesInBlock);
+            indexEntries.emplace_back(entry);
+            lastBlockEndedWithNewline = currentBlockEndedWithNewline;
+        }
+
+        uint startingLines[]{0, 4, 8, 12, 16, 18};
+        uint indexEntryID[]{0, 1, 2, 7, 9, 7};
+        uint startingBlockIDs[]{0, 1, 2, 7, 8, 7};
+        uint expectedLines[]{4, 4, 4, 4, 4, 2};
+
+        // The Extractor works a bit differently than the Indexer. In the Indexer, we decompress whole blocks, whereas
+        // in the Extractor, we decompress chunk-wise (technical reasons, the flush mode Z_BLOCK does not work). One
+        // problem here is, that the skip can be bigger than the decompressed chunk of data and this lead to problems
+        // in the past. To test this, we start with a "lower" index entry and add a skip factor for the LAST test!
+
+        // Please also note, that the last line of the test dataset has NO newline, which can also occur in some source
+        // files. This line will not be found by the tested method here but will be output with a WARNING at the end of
+        // the extraction process.
+
+        for (int i = 0; i < 6; i++) {
+            auto startingLine = startingLines[i];
+            auto lineCount = 4;
+            auto indexEntry = indexEntries[indexEntryID[i]];
+            Extractor extractor(make_shared<PathSource>(fastq),
+                                make_shared<PathSource>(index),
+                                ConsoleSink::create(),
+                                false,
+                                ExtractMode::lines, startingLine, lineCount, 4, true);
+            extractor.calculateStartingLineAndLineCount(); // Needs to be done, otherwise startingLine and linecount are not set.
+            extractor.setSkip(startingLine - indexEntry->startingLineInEntry);
+
+            for (uint64_t j = startingBlockIDs[i]; j < _blockData.size(); j++) {
+                extractor.processDecompressedChunkOfData(_blockData[j], indexEntry->toIndexEntry());
+            }
+                    CHECK(extractor.getStoredLines().size() == expectedLines[i]);
+        }
+    }
+
+    /**
+     * This test will check the case where an incomplete line was printed because the last incomplete line was ignored.
```

Comment:
I don't understand the description. Why should the indexer print an incomplete line?